contracts/multihop/src/utils.rs

```diff
@@ -0,0 +1,25 @@
+use soroban_sdk::{Env, Symbol, Vec};
+
+use crate::storage::Swap;
+
+pub fn verify_operations(env: &Env, operations: &Vec<Swap>, normal_swap: bool) -> Option<Symbol> {
+    if operations.is_empty() {
+        return Some(Symbol::new(env, "operations_empty"));
+    }
+
+    for i in 0..operations.len() - 1 {
+        if normal_swap {
+            if operations.len() > 1
+                && operations.get(i).unwrap().ask_asset
+                    != operations.get(i + 1).unwrap().offer_asset
+            {
+                return Some(Symbol::new(env, "bad_swaps"));
+            }
+        } else if operations.len() > 1
+            && operations.get(i).unwrap().offer_asset != operations.get(i + 1).unwrap().ask_asset
+        {
+            return Some(Symbol::new(env, "bad_swaps"));
+        }
+    }
+    None
+}
```

Comment:
1. Bring back `if operations.is_empty()` to the main functions.
2. Remove that `bool` parameter, create two functions: `verify_swap` and `verify_reverse_swap`.
3. Since swap and reverse_swap will have a different verifications, no need to return anything.
4. Write loops in Rust idiomatic way.

```suggestion
pub fn verify_swap( operations: &Vec<Swap>) {
    for (current, next) in operations.iter().zip(operations.iter().skip(1)) {
        if current.ask_asset != next.offer_asset {
            panic!("Multihop: Swap: Provided bad swap order");
        }
    }
}

pub fn verify_reverse_swap(operations: &Vec<Swap>) {
    for (current, next) in operations.iter().zip(operations.iter().skip(1)) {
        if current.offer_asset != next.ask_asset {
            panic!("Multihop: Reverse swap: Provided bad swap order");
        }
    }
}
```